src/components/Modal/ModalContent.tsx

```diff
@@ -0,0 +1,19 @@
+import type {ReactNode} from 'react';
+import React from 'react';
+
+type ModalContentProps = {
+    /** Modal contents */
+    children: ReactNode;
+
+    /** called after modal content is dismissed */
+    onDismiss: () => void;
+};
+
+function ModalContent({children, onDismiss = () => {}}: ModalContentProps) {
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    React.useEffect(() => () => onDismiss?.(), []);
```

Comment:
trying to understand why do we need this component and call `onDismiss` on mount?